cronback-migration/src/m20230712_205649_add_projects_model.rs

```diff
@@ -0,0 +1,68 @@
+use sea_orm_migration::prelude::*;
+
+#[derive(DeriveMigrationName)]
+pub struct Migration;
+
+#[async_trait::async_trait]
+impl MigrationTrait for Migration {
+    async fn up(&self, manager: &SchemaManager) -> Result<(), DbErr> {
+        manager
+            .create_table(
+                Table::create()
+                    .table(Projects::Table)
+                    .if_not_exists()
+                    .col(ColumnDef::new(Projects::Id).primary_key().string())
+                    .col(
+                        ColumnDef::new(Projects::CreatedAt)
+                            .date_time()
+                            .not_null(),
+                    )
+                    .col(
+                        ColumnDef::new(Projects::LastStatusChangedAt)
+                            .date_time()
+                            .not_null(),
+                    )
+                    .col(
+                        ColumnDef::new(Projects::Status)
+                            .enumeration(
+                                Projects::Status,
+                                vec![
+                                    ProjectStatus::Enabled,
+                                    ProjectStatus::Disabled,
+                                    ProjectStatus::QuotaExceeded,
+                                    ProjectStatus::PendingDeletion,
```

Comment:
I'm not sure I understand the semantics of this. First, what do you mean by enabled/disabled? does that control user access or whether cronback should load it—in scheduler, dispatch, and etc.

For instance, if a project is in `QuotaExceeded`, does that mean it's disabled? what about `PendingDeletion`?

Reply:
What I had in mind is that this status controls whether the scheduler should schedule new runs or not. If a project is "disabled" manually, exceeded the quota or pending deletion, we don't schedule any new runs.

> For instance, if a project is in QuotaExceeded, does that mean it's disabled? what about PendingDeletion?

In the model, you'll find an `is_active` function, only `Enabled` returns true, the rest returns false.

Maybe it should be `scheduling_status`?

Reply:
At high-level, maybe we should have a flag indicating whether its active or not (is_active). That said, we need to define clear semantics to what this means. For instance, can the user create, read, delete triggers from inactive projects? Do we silently ignore those triggers? Potentially, a confusing user experience.

For quota exceeded and pending deletion, they are orthogonal to being active or not, and they might be even logically set simultaneously—project both exceeded the quota and is being deleted. 

In some sense, QuotaExceeded and PendingDeletion represent different things, the first is a status "fact", the second is a state "transitional".

We have a few options:
1. If this is informational only, then something like `StatusDetails` with polymorphic structure is an option (json/grpc-oneof/etc.)
2. If we might use this to take decisions or execute actions, then we might want to split this into a separate flags table, a project can have one or more flags set and each flag can have its own details field.
3. The last option, instead of enum, we add an column for each flag. `has_quota_exceeded`, `is_deleted`, `deletion_requested_at`, `remaining_quota_...`.